test/gbdt/test_gbdt.py

```diff
@@ -86,3 +86,30 @@ def test_gbdt_with_save_load(gbdt_cls, stypes, task_type_and_metric):
         assert (0 <= score <= 1)
     elif task_type == TaskType.MULTICLASS_CLASSIFICATION:
         assert (0 <= score <= 1)
+
+
+def test_xgboost_early_stopping_rounds():
+    dataset: Dataset = FakeDataset(
+        num_rows=30,
+        with_nan=True,
+        stypes=[stype.numerical, stype.categorical],
+        create_split=True,
+        task_type=TaskType.BINARY_CLASSIFICATION,
+        col_to_text_embedder_cfg=TextEmbedderConfig(
+            text_embedder=HashTextEmbedder(8)),
+    )
+    dataset.materialize()
+    gbdt = XGBoost(
+        task_type=TaskType.BINARY_CLASSIFICATION,
+        metric=Metric.ROCAUC,
+    )
+
+    gbdt.tune(
+        tf_train=dataset.tensor_frame,
+        tf_val=dataset.tensor_frame,
+        num_trials=2,
+        num_boost_round=100,
+        early_stopping_rounds=2,
+    )
+
+    assert gbdt.model.best_iteration is not None
```

Comment:
Can we include this test into the test above?

Reply:
Done